pkg/diskmaker/discovery/sync_result.go

```diff
@@ -30,7 +30,7 @@ func newDiscoveryResultInstance(nodeName, namespace, parentObjName, parentObjUID
 			OwnerReferences: []metav1.OwnerReference{
 				{
 					APIVersion: v1alpha1.SchemeGroupVersion.String(),
-					Kind:       "LocalVolumeDiscovery",
```

Comment:
I do not understand why should discovery results object have owneref to pod or daemonset. Shouldn't this be `LocalVolumeDiscovery` object and its UUID.

Reply:
I might be wrong, but how about the following scenario:
1-  User startts Discovery on Node1 and Node2.
2- User updates the Discovery CR to run it only on Node1. 

After Step 2,  daemonset pod running on `Node2` is gone so corresponding discovery result CR will be deleted. 
But if the Discovery Result CR was dependent on `LocalVolumeDiscovery`, then after step 2, daemonset pod will be deleted but the discovery result CR (on Node1) won't be deleted. 

So in short, if we have LocalVolumeDiscovery and its UUID as owner, the results will be deleted only when user deletes the LocalVolumeDiscovery CR complete and not when user updates it.

Reply:
I don't think it is a good idea to depend on ownereref mechanism for cleaning up leftover result objects. We already have established pattern of verifying if node selector includes certain node and may be we should periodically reconcile and delete leftover discovery result objects and use LocalVolumeDiscovery object as owneref.

Reply:
Identifying `leftover discovery result objects` approach might get complicated if we decide to go with multiple `LocalVolumeDiscovery` objects in the future.  Right now, we mandate the user to create only a single `LocalVolumeDiscovery` object. 

With multiple `LocalVolumeDisovery` objects, a node might be removed from one object's nodeSelector, but it still might be part of another object's nodeSelector.

Reply:
@gnufied Updated the owner reference.  LocalVolumeDiscoveryResults have owner reference of LocalVolumeDiscovery. Also deleting any orphan LocalVolumeDiscoveryResult on every reconcile.